### PR TITLE
fixes #4081 App is crash when Backbutton is pressed

### LINKED ISSUE #4081: App is crash when Backbutton is pressed
**Summary:** 

App is crash when we are at Contributions,Favorities page and then backButton is pressed

**Steps to reproduce:** 
->Go to contributions,Favorities
->Backbutton pressed
What did you expect the app to do, and what did you see instead?
App does not crash we backbutton pressed

**Device and Android version:** 
Android Device :Redmi
API :29
**LogCat**
2020-12-07 11:56:58.628 8131-8131/fr.free.nrw.commons.beta E/ACRA: ACRA caught a NullPointerException for fr.free.nrw.commons.beta
    **java.lang.NullPointerException: Attempt to invoke virtual method 'boolean fr.free.nrw.commons.media.MediaDetailPagerFragment.isVisible()' on a null object reference
        at fr.free.nrw.commons.contributions.ContributionsFragment.backButtonClicked(ContributionsFragment.java:610)
        at fr.free.nrw.commons.contributions.MainActivity.onBackPressed(MainActivity.java:241)**
        at android.app.Activity.onKeyUp(Activity.java:3147)
        at android.view.KeyEvent.dispatch(KeyEvent.java:2733)
        at android.app.Activity.dispatchKeyEvent(Activity.java:3430)
        at androidx.core.app.ComponentActivity.superDispatchKeyEvent(ComponentActivity.java:122)
        at androidx.core.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:84)
        at androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:140)
        at androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:558)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
        
**Commons app version:** 
2.13.1-debug-master~b24ed1cf6
**Screen-shots:** 

**Would you like to work on the issue?**
YES
Please let us know whether you want to fix the issue by yourself. If not, anyone can get the issue assigned to them.
YES

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
**Description (required)**
Fixes #4081 
What changes did you make and why?
fixes NullPointerexception in ContributionsFragment and FavoritiesFragment when backButton is pressed

**Tests performed (required)**
Android Device :Redmi Y3
API :29

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 1]
```diff
@@ -97,7 +97,7 @@
 
     private ContributionsListFragment contributionsListFragment;
     private static final String CONTRIBUTION_LIST_FRAGMENT_TAG = "ContributionListFragmentTag";
-    private MediaDetailPagerFragment mediaDetailPagerFragment;
+    static MediaDetailPagerFragment mediaDetailPagerFragment;
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:100]: I don't think this should be static - there should be a `MediaDetailPagerFragment` per `ContributionsFragment`

File: app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java [hunk 2]
```diff
@@ -172,15 +172,21 @@ public void viewPagerNotifyDataSetChanged() {
...
+    //check medaiDetailPage fragment is not null then we check medaiDetail.is Visible or not to avoid NullPointerException
+    if (mediaDetails != null) {
+       if (mediaDetails.isVisible()) {
+        // todo add get list fragment
+        ((BookmarkFragment) getParentFragment()).tabLayout.setVisibility(View.VISIBLE);
+        removeFragment(mediaDetails);
+        setFragment(listFragment, mediaDetails);
+        ((MainActivity) getActivity()).showTabs();
+       } else {
+        ((MainActivity) getActivity()).setSelectedItemId(NavTab.CONTRIBUTIONS.code());
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java:184]: Could avoid code duplication (lines 184 and 188) by combining if statements

File: app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java [hunk 3]
```diff
@@ -172,15 +172,21 @@ public void viewPagerNotifyDataSetChanged() {
   }
 
   public void backPressed() {
-    if (mediaDetails.isVisible()) {
-      // todo add get list fragment
-      ((BookmarkFragment)getParentFragment()).tabLayout.setVisibility(View.VISIBLE);
-      removeFragment(mediaDetails);
-      setFragment(listFragment, mediaDetails);
-      ((MainActivity)getActivity()).showTabs();
+    //check medaiDetailPage fragment is not null then we check medaiDetail.is Visible or not to avoid NullPointerException
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java:175]: nit: typos

File: app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java [hunk 4]
```diff
@@ -172,15 +172,21 @@ public void viewPagerNotifyDataSetChanged() {
...
+    if (mediaDetails != null) {
+       if (mediaDetails.isVisible()) {
+        // todo add get list fragment
+        ((BookmarkFragment) getParentFragment()).tabLayout.setVisibility(View.VISIBLE);
+        removeFragment(mediaDetails);
+        setFragment(listFragment, mediaDetails);
+        ((MainActivity) getActivity()).showTabs();
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java:182]: All code paths end with `((MainActivity) getActivity()).showTabs();` - we could move that to the end to reduce code duplication

**Review by neslihanturan — CHANGES_REQUESTED** [review]
Hey @Prince-kushwaha , some of the reviews are not done yet. Can you make sure you have fixed all reviews? I agree @domdomegg with his reviews.

**Comment by Prince-kushwaha** [comment]
Ok

**Comment by Prince-kushwaha** [comment]
@neslihanturan  see the second pull request of this pull request #4104 becouse this branch is delected

**Comment by Prince-kushwaha** [comment]
@neslihanturan see the another pull request of this pull request #4104 becouse this branch is delect by mistake

**Pull request closed without merging**